core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProjectOrchestrator.scala

```diff
@@ -1,34 +1,72 @@
-import org.broadinstitute.dsde.rawls.serviceperimeter.ServicePerimeterService
-import org.broadinstitute.dsde.rawls.user.UserService.syncBillingProjectOwnerPolicyToGoogleAndGetEmail
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.rawls.dataaccess.SamDAO
+import org.broadinstitute.dsde.rawls.model.{CreateRawlsV2BillingProjectFullRequest, CreationStatuses, ErrorReport, ErrorReportSource, RawlsBillingProject, SamBillingProjectPolicyNames, SamBillingProjectRoles, SamPolicy, SamResourcePolicyName, SamResourceTypeNames, UserInfo}
 import org.broadinstitute.dsde.rawls.{RawlsExceptionWithErrorReport, StringValidationUtils}
 import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
 
-import java.net.URLEncoder
-import java.nio.charset.StandardCharsets.UTF_8
 import scala.concurrent.{ExecutionContext, Future}
 
 
 /**
- * Knows how to provision billing projects with external cloud providers
+ * Knows how to provision billing projects with external cloud providers (that is, implementors of the
+ * BillingProjectCreator trait)
+ *
+ * All billing projects are created following this algorithm:
+ * 1. Pre-flight validation with a billing project creator. Right now, the creator is determined according
+ * to the nature of the billing project creation request and is not client-configurable.
+ * 2. Create the rawls internal billing project record
+ * 3. Post-flight steps; this may include syncing of groups, reaching out to external services to sync state, etc.
+ * This step is delegated to the billing project creator as well.
  */
-class BillingProjectOrchestrator(userInfo: UserInfo, samDAO: SamDAO, gcsDAO: GoogleServicesDAO, billingRepository: BillingRepository)(implicit val executionContext: ExecutionContext) extends StringValidationUtils {
+class BillingProjectOrchestrator(userInfo: UserInfo,
+                                 samDAO: SamDAO,
+                                 billingRepository: BillingRepository,
+                                 googleBillingProjectCreator: BillingProjectCreator,
+                                 bpmBillingProjectCreator: BillingProjectCreator)
+                                (implicit val executionContext: ExecutionContext) extends StringValidationUtils with LazyLogging {
   implicit val errorReportSource = ErrorReportSource("rawls")
 
+  /**
+   * Creates a "v2" billing project, using either Azure managed app coordinates or a Google Billing Account
+   */
   def createBillingProjectV2(createProjectRequest: CreateRawlsV2BillingProjectFullRequest): Future[Unit] = {
+    val billingProjectCreator = createProjectRequest.billingInfo match {
+      case Left(_) => googleBillingProjectCreator
+      case Right(_) => bpmBillingProjectCreator
+    }
+    val billingProjectName = createProjectRequest.projectName
+
     for {
       _ <- validateBillingProjectName(createProjectRequest.projectName.value)
```

Comment:
What is the value of the for comprehension if none of the values are carried through, and we aren't returning anything? Wouldn't just a foreach be more clear for that?